src/core/handle.ml

```diff
@@ -185,7 +199,54 @@ let handle_cmd : sig_state -> p_command -> sig_state * proof_data option =
       (* Actually add the symbol to the signature. *)
       out 3 "(symb) %s ≔ %a\n" x.elt pp_term t;
       let d = if op then None else Some(t) in
-      let ss = Sig_state.add_symbol ss e Defin x a impl d in
+      (fst (Sig_state.add_symbol ss e Defin x a impl d), None)
+  | P_inductive(e, id, a, l)     ->
+      (* Add the inductive type in the signature *)
+      let (ss, sym_typ) = handle_symbol ss e Defin id [] a in
+      (* Add the constructors in the signature.  *)
+      let add_cons :
+        sig_state -> (ident * p_term) list-> sig_state * sym list =
+        fun ss l ->
+        let rec aux acc ss l =
+          match l with
+          | []   -> ss, acc
+          | (id, a)::q ->
+              let (ss, acc) = aux acc ss q in
+              let (ss, sym_cons) = handle_symbol ss e Const id [] a in
+              ss, sym_cons::acc
+        in
+        aux [] ss l
+      in
+      (*let add_cons :
+            sig_state * sym list -> ident * p_term -> sig_state * sym list
+        = fun(ss, cons_list) (id, a) ->
+        let (ss, sym_cons) = handle_symbol ss e Const id [] a in
+        (ss, sym_cons::cons_list)
+      in
+      let (ss, cons_list) = List.fold_left add_cons (ss, []) l in *)
+      let (ss, cons_list) = add_cons ss l in
+      (* Compute the induction principle *)
+      let ind_typ = Inductive.principle ss cmd.pos sym_typ cons_list in
+      (* Add the induction principle in the signature *)
+      let ind_name = Pos.make cmd.pos ("ind_" ^ id.elt) in
+      if StrSet.mem id.elt !(ss.signature.sign_idents) then
+        Sign.add_ident ss.signature ind_name.elt;
+      let (ss, sym_ind) =
+        Sig_state.add_symbol ss e Defin ind_name ind_typ [] None
+      in
+      (* Compute the rules associated with the induction principle *)
+      let rs =
+        Inductive.ind_rule sym_typ.sym_name ("ind_"^id.elt) ind_typ
+          cons_list
+      in
+      let (ss, ind_rules) = handle_rules ss rs       in
+      let ind_rules = List.map (fun (_,r) -> r.elt) ind_rules in
+      (* Add the rules of induction principle in the signature *)
+      let just_add_rule : sym -> rule -> unit = fun sym r ->
+        sym.sym_rules := r :: !(sym.sym_rules) in
+      List.iter (just_add_rule sym_ind) ind_rules;
```

Comment:
we can replace those 3 lines by just `sym.sym_rules := ind_rules @ !(sym.sym_rules) `